pixutils/conv/raw.py

```diff
@@ -114,7 +114,8 @@ def demosaic(data: npt.NDArray[np.uint16], pattern: BayerPattern) -> npt.NDArray
     h, w = data.shape
 
     # Separate the components from the Bayer data to RGB planes
-    rgb = np.zeros((h, w, 3), dtype=data.dtype)
+    # Use a larger data type to avoid overflow when debayering 16 bit images
+    rgb = np.zeros((h, w, 3), dtype=np.uint32)
```

Comment:
I don't think this is quite right.

Maybe `p = rgb[..., plane].view(dtype=np.uint32)` a bit below in the function?

Reply:
Also, I suggest doing a perf test with the old and new code. I think the change you suggest doubles the size of multiple buffers, which might easily have a visible perf impact.

Reply:
Is there an easy/existing way to measure the perf of these? or did you have something in mind?

If we can just make the calculation in 32 bit and keep 16 bit buffers/data accesses I'm fine with that ... is that what your proposal will do ?

Reply:
> Is there an easy/existing way to measure the perf of these? or did you have something in mind?

I think just a simple loop calling the conv, and measuring the time should do it. That's easy. I don't have anything ready for it.

> If we can just make the calculation in 32 bit and keep 16 bit buffers/data accesses I'm fine with that ... is that what your proposal will do ?

I'm not sure, but I hope so =)

Reply:
Also, please run the tests. This seems to break the test_conv.py. I assume it's only the SRGGB16 test, but that needs to be verified and the test data updated.

Reply:
> ```
>     p = rgb[..., plane].astype(np.uint32)
> ```
> 
> gets through the interpreter ....

I think that makes a copy, so it's not ok. It's better to convert the rgb array. But... It feels silly to convert it to 32, as we never store more than 16 bits per elemenet.

Reply:
`p = rgb[..., plane].astype(dtype=np.uint32, copy=False)`